packages/shorebird_cli/lib/src/shorebird_engine_config.dart

```diff
@@ -0,0 +1,39 @@
+// A reference to a [EngineConfig] instance.
+import 'package:scoped/scoped.dart';
+
+final engineConfigRef = create(() => const EngineConfig.empty());
+
+// The [EngineConfig] instance available in the current zone.
+EngineConfig get engineConfig => read(engineConfigRef);
+
+class EngineConfig {
+  EngineConfig({
+    this.localEngineSrcPath,
+    this.localEngine,
+    this.localEngineHost,
+  }) {
+    final args = [localEngineSrcPath, localEngine, localEngineHost];
+    final allArgsAreNull = args.every((arg) => arg == null);
+    final allArgsAreNotNull = args.every((arg) => arg != null);
+    if (!allArgsAreNull && !allArgsAreNotNull) {
+      // Only some args were provided, this is invalid.
+      throw ArgumentError(
```

Comment:
looks like we're missing a test for this?

Reply:
Done